File: frontend/src/types/invitation.ts
/**
 * Invitation-related types matching backend schemas.
 */

export interface InviteRequest {
  email: string;
}

export interface InvitationResponse {
  invitation_id: string;
  poll_id: string;
  inviter_id: string;
  invitee_id: string;
  invitee_email: string;
  invitee_username: string;
  status: 'active' | 'revoked';
  created_at: string;
  revoked_at: string | null;
}

export interface InvitationListResponse {
  invitations: InvitationResponse[];
  total: number;
}

File: backend/app/schemas/invitation.py
"""
Pydantic v2 schemas for invitation operations.
"""

from datetime import datetime
from typing import Optional

from pydantic import BaseModel, EmailStr, Field


class InviteRequest(BaseModel):
    """Payload for ``POST /api/v1/polls/:id/invitations``."""

    email: EmailStr = Field(
        ...,
        description="Email address of the user to invite.",
    )


class InvitationResponse(BaseModel):
    """Single invitation representation."""

    invitation_id: str
    poll_id: str
    inviter_id: str
    invitee_id: str
    invitee_email: str
    invitee_username: str
    status: str  # "active" or "revoked"
    created_at: datetime
    revoked_at: Optional[datetime] = None


class InvitationListResponse(BaseModel):
    """List of invitations for a poll."""

    invitations: list[InvitationResponse]
    total: int
